pkg/pattern/ingester_querier.go

```diff
@@ -44,10 +45,24 @@ func NewIngesterQuerier(
 }
 
 func (q *IngesterQuerier) Patterns(ctx context.Context, req *logproto.QueryPatternsRequest) (*logproto.QueryPatternsResponse, error) {
-	_, err := syntax.ParseMatchers(req.Query, true)
+	expr, err := syntax.ParseLogSelector(req.Query, true)
 	if err != nil {
 		return nil, httpgrpc.Errorf(http.StatusBadRequest, err.Error())
 	}
+	var queryErr error
+	expr.Walk(func(treeExpr syntax.Expr) {
```

Comment:
I put this here to guard against anyone passing in more complex queries to this API. Only allowing a subset of LogQL is a bit of an odd use case so I'd appreciate any input on whether there's a better way to do this.